### PR TITLE
Remove context.TODO() from client_go method calls

### PR DESCRIPTION
**Pull request by slashpai**
As in title [MON-1251](https://issues.redhat.com/browse/MON-1251)

This is first PR to remove context.TODO() and replace with real context in tests

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by jan--f — CHANGES_REQUESTED** [review]
Just a nit, otherwise lgtm.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,5 @@ tmp/
 # purpose is to track the last target execution time to evaluate, whether the
 # container needs to be rebuilt
 .hack-*-image
+
+.idea
```
[the remark this answers is not in the document]
Comment (slashpai) [reply .gitignore]: Thank you didn't know about it :) I have removed that commit

File: test/e2e/framework/client.go [hunk 2]
```diff
@@ -47,8 +47,9 @@ func NewPrometheusClientFromRoute(
 	routeClient routev1.RouteV1Interface,
 	namespace, name string,
 	token string,
+	ctx context.Context,
```
Comment (dgrisonnet) [inline test/e2e/framework/client.go:50]: nit: let's be consistent and put the context parameter in second place after the client

**Pull request merged**